### DISCUSSION TITLE
Struggling to use actions to run Xcode XCUITest (automated UI tests)

### DISCUSSION THREAD

**Discussion by simonmcl** [community/community#65667] [Actions]
### Select Topic Area

General

### Body

Not really sure if this is a question, feedback or general discussion. Looking to hear other peoples experiences, but also hopefully find some resolutions

I've been using actions now for a while to setup pipelines doing multiple things. Building my Xcode project, archiving it, deploy to testflight, extract dsym's for my crash reporting tool, running localisation scripts, auto incrementing build number and committing it back. It was a learning curve for sure, but its been amazing

I recently added a bunch of XCUITests to my project, have them running locally fine (app has been through multiple rounds of external beta testing too). Trying to run them on github actions ... almost nothing works. I'm having to add `sleep(10)` everywhere to let the app catch up with the test runner because all the inbuilt/default timeouts keep failing. Even simple stuff like using `typeText(...)` to enter something into a textfield, where a validator will run a basic check and enable a button, requires adding a long sleep in between the text entry and the button tap. Anything with even a hint of complex logic is triggering "app hanging" warnings and bizarre crash reports in my crash reporter (none can be reproduced on even very old real devices).

My app is not small, it has +15 swift packages in it, but its also not a massive app. During the onboarding it hovers around 100mb of RAM. I'm also using the latest Xcode which means I needed to use `macos-13` in github actions, which I think is still listed as Beta. Not sure if the beta ones get less resources and that might be the issue? but actions has a problem building the project with a lower version, so I can't downgrade

Anyone else have any similar experiences or thoughts?

**Comment by amyworrall** [comment]
I don't have a solution, but I'm following along because I'm about to start integrating UI tests into actions

**Reply by simonmcl** [reply]
@amyworrall any luck?
